### PR TITLE
[APK] APIUtill and related flows refactor for APK

### PR DESCRIPTION
**Pull request by chameerar**
$subject

### CODE REVIEW

File: apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/PasswordResolverFactory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    private PasswordResolverFactory() {
+    }
+
+    /**
+     * Read values from APIManagerConfiguration.
+     *
+     * @throws APIManagementException
+     */
+    //TODO use proper init strategy with configs
```
Comment (lasanthaS) [inline apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/PasswordResolverFactory.java:37]: Remove commented-out code.

File: apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/config/APIMConfigServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    @Override
+    public void updateExternalStoreConfig(String organization, String externalStoreConfig)
+    {
+    }
+
+    @Override
+    public String getExternalStoreConfig(String organization) throws APIManagementException {
+
+        //TODO registry implementation
```
Comment (lasanthaS) [inline apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/config/APIMConfigServiceImpl.java:59]: Remove commented-out code.

File: apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    /**
+     * Check whether given url is a HTTPS url.
+     *
+     * @param url url to check
+     * @return true if the given url is HTTPS, false otherwise
+     */
+    private static boolean isHttpsURL(String url) {
+
```
Comment (lasanthaS) [inline apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java:162]: Remove unnecessary blank line

File: apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    /**
+     * Check whether given url is a WS url.
+     *
+     * @param url url to check
+     * @return true if the given url is WS, false otherwise
+     */
+    private static boolean isWebSocketURL(String url) {
+
```
Comment (rashm1n) [inline apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java:173]: Unnecessary blank line, also in L184

File: apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
+/*
+ *
+ *   Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (rashm1n) [inline apk/org.wso2.apk.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/apk/apimgt/rest/api/publisher/v1/common/mappings/EnvironmentMappingUtil.java:3]: Change data on licensing header. Check other files as well.

File: apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/config/APIMConfigServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    public APIMConfigServiceImpl() {
+        systemConfigurationsDAO = SystemConfigurationsDAO.getInstance();
+    }
+
+    @Override
+    public void addExternalStoreConfig(String organization, String externalStoreConfig) {
+    }
+
+    @Override
+    public void updateExternalStoreConfig(String organization, String externalStoreConfig)
```
Comment (rashm1n) [inline apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/config/APIMConfigServiceImpl.java:52]: Fix formatting

File: apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/PasswordResolverFactory.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * This method will take hardcoded class name for password resolver from api-manager.xml file and will return
+     * an instance of PasswordResolver.
+     *
+     * @return PasswordResolver instance.
+     */
+    public static PasswordResolver getInstance() {
+
+//        return passwordResolver;
```
Comment (rashm1n) [inline apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/PasswordResolverFactory.java:72]: Is this comment needed ?

File: apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/DefaultPasswordResolverImpl.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+package org.wso2.apk.apimgt.impl;
+
+import org.wso2.apk.apimgt.api.PasswordResolver;
+
+public class DefaultPasswordResolverImpl implements PasswordResolver {
+
+    @Override
+    public String getPassword(String encryptedPassword) {
+
```
Comment (rashm1n) [inline apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/DefaultPasswordResolverImpl.java:27]: Unnecessary blank line

**Pull request merged**